packages/swapper/src/swappers/JupiterSwapper/utils/helpers.ts

```diff
@@ -253,25 +260,29 @@ export const createSwapInstructions = async ({
     {},
   )
 
-  const { instruction: feeAccountInstruction, tokenAccount } =
-    await getFeeTokenAccountAndInstruction({
-      feePayerPubKey: new PublicKey(sendAddress),
-      buyAssetReferralPubKey,
-      sellAssetReferralPubKey,
-      programId: new PublicKey(JUPITER_AFFILIATE_CONTRACT_ADDRESS),
-      instructionData,
-      buyTokenId: buyAssetAddress,
-      sellTokenId: sellAssetAddress,
-      connection: adapter.getConnection(),
-    })
+  const { instruction: feeAccountInstruction, tokenAccount } = !isBuyAssetToken2022
+    ? await getFeeTokenAccountAndInstruction({
```

Comment:
What a catch, it should actually be a blocking, fixed in [a784a1b](https://github.com/shapeshift/web/pull/8365/commits/a784a1ba34bda2a8736c523a2ecb70bf022c21b8)

I think it's legacy from my rubber duck phase while attempting to fix those token 2022 edge cases, it's already handled under the hood by getFeeTokenAccountAndInstruction, if none of the assets are SPL Tokens, then it returns undefined

Also I noticed a bug with referral account if you swap to token2022 that I fixed in [9931ddf](https://github.com/shapeshift/web/pull/8365/commits/9931ddf7ecd7fec9545157566fc829e16b4bbfd2)

Unfortunately this bug is forcing us to use `useSharedAccounts` (which is what Jupiter uses under the hood), I feel like it's less safe, but in the end it might be better for the user anyway because fee amounts are way less than before

https://solscan.io/tx/3vScHkxZwMR2ofSCkuHnHVjCpEsfvdR9d8iseyAGT6txQnm8sBKjSxMsBxaToAkk1us71SdLKK3wRwBVAZ52ArCv